config_test.go

```diff
@@ -226,3 +240,143 @@ func TestLoadFromValueUnmarshaler(t *testing.T) {
 	require.Equal(t, 10, s.Age)
 	require.Zero(t, s.Ignored)
 }
+
+type backendCalled struct {
+	store  map[string]string
+	called int
+}
+
+func (m *backendCalled) Get(ctx context.Context, key string) ([]byte, error) {
+	m.called++
+	data, ok := m.store[key]
+	if !ok {
+		return nil, backend.ErrNotFound
+	}
+
+	return []byte(data), nil
+}
+
+func (backendCalled) Name() string {
+	return "backendCalled"
+}
+
+type backendNotCalled struct {
```

Comment:
This type is exactly the same as the one above, why not use only one and make the name customizable ?